#! /bin/sh /usr/share/dpatch/dpatch-run
## 02-drb-client-hang.dpatch by  <seb@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes #572

@DPATCH@
diff -urNad jruby-ut0.9-0.9.8/lib/ruby/1.8/drb/drb.rb /tmp/dpep.g2oRMf/jruby-ut0.9-0.9.8/lib/ruby/1.8/drb/drb.rb
--- jruby-ut0.9-0.9.8/lib/ruby/1.8/drb/drb.rb	2007-03-05 15:44:08.000000000 -0800
+++ /tmp/dpep.g2oRMf/jruby-ut0.9-0.9.8/lib/ruby/1.8/drb/drb.rb	2007-03-22 16:39:19.000000000 -0700
@@ -951,7 +951,7 @@
     # Check to see if this connection is alive.
     def alive?
       return false unless @socket
-      if IO.select([@socket], nil, nil, 0)
+      if IO.select([@socket], nil, nil, 0.05)
 	close
 	return false
       end
